### ISSUE TITLE
[CI] SearchableSnapshotActionIT testConvertingSearchableSnapshotFromFullToPartial failing

### ISSUE THREAD

**Issue by ywangd** [elastic/elasticsearch#73706]
It's the first failure in last 7 days. But the error message looks legit enough.

**Build scan:**
https://gradle-enterprise.elastic.co/s/kkeggcanhv3ks/tests/:x-pack:plugin:ilm:qa:multi-node:javaRestTest/org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT/testConvertingSearchableSnapshotFromFullToPartial

**Reproduction line:**
`./gradlew ':x-pack:plugin:ilm:qa:multi-node:javaRestTest' --tests "org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT.testConvertingSearchableSnapshotFromFullToPartial" -Dtests.seed=BA9C66979C640768 -Dtests.locale=en-GB -Dtests.timezone=America/Danmarkshavn -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT&tests.test=testConvertingSearchableSnapshotFromFullToPartial

**Failure excerpt:**
```
org.elasticsearch.client.ResponseException: method [DELETE], host [http://[::1]:43375], URI [restored-myindex-lfqc], status line [HTTP/1.1 400 Bad Request]
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"The provided expression [restored-myindex-lfqc] matches an alias, specify the corresponding concrete indices instead."}],"type":"illegal_argument_exception","reason":"The provided expression [restored-myindex-lfqc] matches an alias, specify the corresponding concrete indices instead."},"status":400}

  at __randomizedtesting.SeedInfo.seed([BA9C66979C640768:E9794DE67AE7774F]:0)
  at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:330)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:296)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:270)
  at org.elasticsearch.test.rest.ESRestTestCase.wipeSearchableSnapshotsIndices(ESRestTestCase.java:815)
  at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:574)
  at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:342)
  at jdk.internal.reflect.GeneratedMethodAccessor13.invoke(null:-1)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:834)

```

**Comment by ywangd** [comment]
Failed on 7.13 as well: https://gradle-enterprise.elastic.co/s/4skjos437lu3a
But the error message is different
```

org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT &gt; testConvertingSearchableSnapshotFromFullToPartial FAILED | &nbsp;
-- | --
&nbsp; | org.elasticsearch.client.ResponseException: method [DELETE], host [http://127.0.0.1:32881], URI [restored-myindex-qqrd], status line [HTTP/1.1 404 Not Found] | &nbsp;
&nbsp; | {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [restored-myindex-qqrd]","index_uuid":"_na_","resource.type":"index_or_alias","resource.id":"restored-myindex-qqrd","index":"restored-myindex-qqrd"}],"type":"index_not_found_exception","reason":"no such index [restored-myindex-qqrd]","index_uuid":"_na_","resource.type":"index_or_alias","resource.id":"restored-myindex-qqrd","index":"restored-myindex-qqrd"},"status":404} | &nbsp;
&nbsp; | at __randomizedtesting.SeedInfo.seed([22637201A4B9656C:71865970423A154B]:0)

```

**Comment by dakrone** [comment]
Another failure on 7.13: https://gradle-enterprise.elastic.co/s/g7kz7jgnzfvz4

```
org.elasticsearch.client.ResponseException: method [DELETE], host [http://[::1]:60869], URI [restored-myindex-afys], status line [HTTP/1.1 400 Bad Request] | &nbsp;
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"The provided expression [restored-myindex-afys] matches an alias, specify the corresponding concrete indices instead."}],"type":"illegal_argument_exception","reason":"The provided expression [restored-myindex-afys] matches an alias, specify the corresponding concrete indices instead."},"status":400}
```

**Comment by tlrx** [comment]
In #73650 I modified EsRestCase in order to delete searchable snapshots indices in the after test clean up logic. This additional clean up logic made this test to fail because sometimes the index was deleted by ILM at the time it is executed. I merged #73753 to ignore the case where the index is not found (that's ok - it does not exist anymore and that's all we want) and I opened #73888 to deal with the situation where the index is deleted but converted as an alias.
